core/dbt/include/global_project/macros/etc/configs.sql

```diff
@@ -0,0 +1,21 @@
+{% macro set_sql_header(config) -%}
+  {{ config.set('sql_header', caller()) }}
+{%- endmacro %}
+
+
+{% macro should_full_refresh() %}
+  {% set config_full_refresh = config.get('full_refresh') %}
+  {% if config_full_refresh is none %}
+    {% set config_full_refresh = flags.FULL_REFRESH %}
+  {% endif %}
+  {% do return(config_full_refresh) %}
+{% endmacro %}
+
+
+{% macro should_store_failures() %}
+  {% set config_store_failures = config.get('store_failures') %}
+  {% if config_store_failures is none %}
+    {% set config_store_failures = flags.STORE_FAILURES %}
+  {% endif %}
+  {% do return(config_store_failures) %}
+{% endmacro %}
```

Comment:
is this used outside of the test materialization? for me this `configs.sql` is the once file i feel iffy about existing, but I see where you're coming from.

Reply:
To me, `should_store_failures()` feels very closely related to `should_full_refresh()`. They have almost identical logic. The `should_full_refresh()` macro is used by both the incremental model materialization + seeds. Do you think it would make more sense to move `configs.sql` to `macros/materializations/` instead of `macros/etc`?

Reply:
I've decided, yes, it would make more sense to have these in `macros/materializations/configs.sql`